docs/setup-cluster/deploy-cluster/slurm/hpc-security-considerations.rst

```diff
@@ -0,0 +1,77 @@
+.. _hpc-security-considerations:
+
+######################################
+ HPC Launcher Security Considerations
+######################################
+
+User authentication in Determined is enforced using :ref:`users`. Determined sends requests on
+behalf of those authenticated users to the HPC launcher which then interacts with the underlying
+workload manager to submit and control jobs via the user agent configured for the user account. A
+Determined administrator must configure a Determined user's agent to enable them to launch Slurm/PBS
+jobs.
+
+There are several security issues that should be considered when deploying the launcher:
+
+#. The specified ``resource_manager.user_name`` should be a unique, non-privileged user with
+   authorization to interact with the deployed workload manager (Slurm/PBS). The launcher executes
+   as a service using the configured ``resource_manager.user_name`` and
+   ``resource_manager.group_name`` as specified in the :ref:`slurm/pbs section
+   <cluster-configuration-slurm>` of the cluster configuration. The launcher can also be run as the
+   ``root`` user but with the corresponding reduction in security isolation.
+
+#. The launcher installs the necessary ``sudoers(5)`` configuration with the file
+   ``/etc/sudoers.d/zz_launcher`` to enable the specified ``resource_manager.user_name`` to perform
+   the following actions:
+
+   -  Change the ownership of a directory tree to another user (from ``resource_manager.user_name``
+      to the Determined user before the job starts, and from the Determined user back to the
+      ``resource_manager.user_name`` after completion).
+
+   -  Enable the execution Slurm/PBS commands on behalf of the Determined user to submit and control
+      their jobs.
+
+#. The set of users ``sudo`` authorizes for Slurm/PBS launch is controlled by
+   ``resource_manager.sudo_authorized``. The default value is ``ALL``. The configuration of users
```

Comment:
Maybe I missed it, but I don't see an example of how the user names would be specified for "sudo_authorized".  For example, is it a space-separated list or a comma-separated list?

Showing an example of what "sudo_authorized" would look like, along with how the generated zz_launcher file would look like, would help, but I don't know if there is some kind of documentation restriction on showing user names, such as "johndoe", "sallysmith", etc.

Reply:
It is a comma-separated member list. For my testing, I can do "sudo_authorized=ALL, !hera, !pluto", hera and pluto are the names of my dogs :). 

Runas_List ::= Runas_Member |
Runas_Member ',' Runas_List
Runas_Member ::= ‘!’* user name |
‘!’* #user-ID |
‘!’* %group |
‘!’* %#group-ID |
‘!’* %:nonunix_group |
‘!’* %:#nonunix_gid |
‘!’* +netgroup |
‘!’* Runas_Alias |
‘!’* ALL